internal/api/identity.go

```diff
@@ -65,3 +68,36 @@ func (a *API) DeleteIdentity(w http.ResponseWriter, r *http.Request) error {
 
 	return sendJSON(w, http.StatusOK, map[string]interface{}{})
 }
+
+func (a *API) LinkIdentity(w http.ResponseWriter, r *http.Request) error {
+	rurl, err := a.GetExternalProviderRedirectURL(w, r)
+	if err != nil {
+		return err
+	}
+	return sendJSON(w, http.StatusOK, map[string]interface{}{
+		"url": rurl,
+	})
```

Comment:
Should we return the oauth provider url in the response body as JSON or should we use a `http.Redirect`? 

It depends on how we want to implement it on the client side. If it's the former, then the url is parsed from the response body JSON. If it's the latter, then the url is parsed from the `Location` header.

Reply:
I guess this is the issue with the existing `/authorize` as well. The problem with HTTP redirects is that `fetch` does not like them at all.

I would do a default HTTP 302 here but if there's a query param `skip_browser_redirect` then return it like JSON with HTTP 200 instead. We already do this for SSO login though there it's a POST so you do send a body.